### ISSUE TITLE
"List All Topics" has duplicates.

### ISSUE THREAD

**Issue by BrentAllsop** [the-canonizer/canonizer.2.0#368]
When you select the "List All Topics" button on the main page (under the top 10 topics) there are duplicates in the list.

Like topic #42 "Human Accomplishment" which appears twice in the list.

**Comment by BrentAllsop** [comment]
Allso, when general is selected, other topics show up, like the "Communicating Ineffable", topic number 102 is in /terminology/, but shows up in the general list.

**Comment by rupalichavan9860** [comment]
@BrentAllsop @gautamv16 I have also tested this."/general/" is listing all topics.
Canonized list for drop down should have one more item "All" for all topics and "/general/" should display only topics under "/general/" namespace

**Comment by BrentAllsop** [comment]
An "All" choice.  Good idea.

**Comment by BrentAllsop** [comment]
I'm thinking this choice should be at the bottom of this list?
It will require LOTS of canonization CPU time, and in most cases be way to long of a list?

**Comment by rupalichavan9860** [comment]
@BrentAllsop @gautamv16  "Communicating Ineffable", topic number 102 is in /terminology/, but shows up in the general list because this topic has been moved from /general/ namespace to /terminology/.
And now we can see this topic in both namespaces(/general/ and /terminology/)
Please refer attached screenshot
![image](https://user-images.githubusercontent.com/40416610/56894867-dd361280-6aa4-11e9-9963-c85846fa1c1f.png)

**Comment by BrentAllsop** [comment]
Yes, the topics that show up, should only be the values determined by the as-of selector, right?

**Comment by rupalichavan9860** [comment]
@gautamv16 Same issue exists on "List of supported camps" page.
Please refer https://staging.canonizer.com/topic-history/239-123456789012345678901234567890 for more details
I created the topic "123456789012345678901234567890" in /general/ namespace and then moved to /sandbox testing/ but if I go to "List of supported camps" page, I can see the same topic in both namespaces
Please refer attached screenshot
![image](https://user-images.githubusercontent.com/40416610/61195103-a3f46280-a6e3-11e9-849a-0975fedbbce5.png)

![image](https://user-images.githubusercontent.com/40416610/61195122-bff80400-a6e3-11e9-9865-bd865eb32751.png)

Expected Result : 
Topic should display only under actual namespace list

**Comment by gautamv16** [comment]
@rupalichavan9860 Pls Test

**Comment by rupalichavan9860** [comment]
@gautamv16 Tested . I am facing two issues
1. I just moved topic https://staging.canonizer.com/topic-history/241-Testing-3-jan-2020 from /sandbox testing/->/sandbox/->/general/->/sandbox testing/ and now live one is under /sandbox testing/ namespace but still I can see this topic under both /sandbox testing/ and /general/ on canonizer main page
![image](https://user-images.githubusercontent.com/40416610/72034865-41f46e80-32bc-11ea-9de0-78f45a85b141.png)

![image](https://user-images.githubusercontent.com/40416610/72034931-6bad9580-32bc-11ea-9859-6cb354e41551.png)

2. On User supports page namespace drop down is not working properly
When I am selecting drop down it is always displaying previously selected namespace name (but correct namespace in URL)
Please refer attached screenshot
Previously I have selected sandbox testing and when I am selecting general page is not getting refreshed. 
General having namespace id = 1
![image](https://user-images.githubusercontent.com/40416610/72035149-0f974100-32bd-11ea-9744-461bda866b50.png)

**Comment by gautamv16** [comment]
@rupalichavan9860 Pls test

**Comment by rupalichavan9860** [comment]
@gautamv16 Tested. 
Point 2 is working fine.
Point 1 : 
General is still displaying some topics from sandbox namespace
![image](https://user-images.githubusercontent.com/40416610/72247201-646aec80-361a-11ea-9c49-2cf053677e0e.png)

![image](https://user-images.githubusercontent.com/40416610/72247271-8a908c80-361a-11ea-8fe6-b1dafb5230f2.png)

**Comment by gautamv16** [comment]
@rupalichavan9860 Pls Test

**Comment by rupalichavan9860** [comment]
@gautamv16 Tested. Count of topics on Browse page and Canonized list for is different for general and sandbox namespaces
I am not able to find exact mismatch for general namespace ,as the topic count is high.
Please refer following sandbox namespace scenario for more details : 
I can see three extra topics under /sandbox/ namespace on browse page .Please refer following screenshot
Note : From this 3 topics  "Testing 3 jan 2020" has been moved to /sandbox testing/ but still listing under /sandbox/ (new issue)
![image](https://user-images.githubusercontent.com/40416610/72701442-561a5480-3b75-11ea-8778-ace4123ea4e6.png)

![image](https://user-images.githubusercontent.com/40416610/72701539-a1346780-3b75-11ea-8b2f-43fb98168b9d.png)

**Comment by gautamv16** [comment]
@rupalichavan9860 after investigation found that its correct as on home page its with filter > 0.01 and on browse page all are listing so its not an issue. So as per me its not an issue and fixed the issue of moved topic from sandbox to sandbox testing topic.

**Comment by rupalichavan9860** [comment]
@gautamv16 I am still facing issues. 
1. Canonized list for still not listing all topics under general namespace.
Following are some test topics that I am not able to see in Canonized list for but present on browse page
https://staging.canonizer.com/topic/238-test-all-topic/1
https://staging.canonizer.com/topic/236-test-new-topic/1
https://staging.canonizer.com/topic/246-test-topic/1

2. Also Some topics I can see only after applying filter as of (yy/mm/dd) : 1/23/2020, 10:00:00 PM
which are not listing for default filter(refer screenshot)

3. Also some topics displaying twice(refer screenshot)
![image](https://user-images.githubusercontent.com/40416610/73008381-7461a800-3e34-11ea-918d-ce571581f9a9.png)

**Comment by gautamv16** [comment]
Rupali,

Topic you are talking about have no support and will not come on home page
without filter change. So comparison with browse page will never match with
home page as there are different conditions on home page and browse page is
all about without conditions. So this not an issue.

Will check for duplicate, that also need to check what you did either u
have same name twice.

**Comment by rupalichavan9860** [comment]
@gautamv16 I have checked and topics I have listed have support.
From above 3 topics 2 are created by you and you are the supporter and 1 topic is created by me and I am the supporter
Please check attached screenshot
![image](https://user-images.githubusercontent.com/40416610/73089174-b4d82900-3efb-11ea-99e1-6a57b4fc4262.png)

![image](https://user-images.githubusercontent.com/40416610/73089237-d76a4200-3efb-11ea-92cb-13528b42429d.png)

![image](https://user-images.githubusercontent.com/40416610/73089314-fbc61e80-3efb-11ea-93a4-c8d531c40d39.png)


Can you please check once again and let me know if I am missing something

**Comment by gautamv16** [comment]
@rupalichavan9860 rupali after so many hours debugging for days found issue with load more topic offset as we have multiple enteries for single topic and when we are providing offset after first load of 30 topic it taking first 30 id not topic so in that way few topic was missing from home page now i have removed load more topic AND all topic as per default namespace will come once.

**Comment by rupalichavan9860** [comment]
@gautamv16 I have tested and looks good now.Multiple entries issue is resolved now but still I am working on count of topics on browse and canonizer main page.If this will work, I will close the issue

**Comment by rupalichavan9860** [comment]
@gautamv16 Tested and verified.Count of topics are matching now so closing this issue
